utopia-remix/app/routes/internal.projects.$id.access.tsx

```diff
@@ -18,13 +17,10 @@ export async function action(args: ActionFunctionArgs) {
 }
 
 export async function handleChangeProjectAccess(req: Request, params: Params<string>) {
-  const user = await requireUser(req)
+  await requireUser(req)
```

Comment:
wait! by removing the user ownership check this means that any user can change the project access of any other user's projects 😱 (or am i hallucinating?) (if that's the case this should **absolutely** tested)

Reply:
ah yeah I just checked: there should be absolutely a test for making sure bob cannot change alice's project access

Reply:
no, since the validator is checking it:
```
export async function action(args: ActionFunctionArgs) {
  return handle(args, {
    POST: {
      handler: handleChangeProjectAccess,
      validator: validateProjectAccess(UserProjectPermission.CAN_MANAGE_PROJECT),
    },
  })
}
```
this was the actual concept of the validator - to take the functionality of access check from the handler

Reply:
Sure. Can you add a test for this then? the validator I mean.

_Note for future selves: it'd be good to have full tests for the whole actions, not just handlers._

Reply:
in any case I added a test for the entire `action` that checks for unauthorized access (though again it mainly checks the validator logic, and not this specific route)

Reply:
yes, I started testing entire actions (for the `/project` endpoint and the `/access` endpoint), this should definitely be the standard now (since it checks access validation as well)

Reply:
(fixed)